### ISSUE TITLE
Additional features for pl.spatial function

### ISSUE THREAD

**Issue by giovp** [theislab/scanpy#1049]
- [ ] Additional function parameters / changed functionality / changed defaults?
- [ ] New analysis tool: A simple analysis tool you have been using and are missing in `sc.tools`?
- [x] New plotting function: A kind of plot you would like to see in `sc.pl`?
- [ ] External tools: Do you know an existing package that should go into `sc.external.*`?
- [ ] Other?

Here's some features that we should add after the scatterplot module is refactored:

- [ ] Add [scalebar](https://pypi.org/project/matplotlib-scalebar/) on the tissue image as suggested on [Twitter](https://twitter.com/ZanotelliVRT/status/1224996059434430464?s=20) . 
- [ ] Change default to “plot `hires` if available, else `lowres` if available, else nothing” . 
- [ ] Add alpha smoothing for continuous features as in [Seurat](https://satijalab.org/seurat/v3.1/spatial_vignette.html) . 

I'll modify this issue if other ideas pop up.

**Comment by ivirshup** [comment]
Does the refactor of the scatterplot module include moving the code for spatial plotting out of `sc.pl.embedding`?

**Comment by giovp** [comment]
Not sure what's @flying-sheep opinion on this. So far, `pl.spatial` is essentially a wrapper of `sc.pl.embedding` with a couple of additions:
* function to process images and plot them 
* use of "circles" (from PatchCollection) instead of "scatter"

**Comment by flying-sheep** [comment]
Yeah, we added a bit more to what’s already too many options. No idea how to nicely deprecate and replace it by something better, and that didn’t change by those additions.

I think the best would be to extract the image-below-things to `sc.pl.spatial`, but leave the option to do circles instead of markers.

**Comment by ivirshup** [comment]
I don't think those options have been [documented in a release](https://scanpy.readthedocs.io/en/stable/api/scanpy.pl.embedding.html#scanpy.pl.embedding).  Can we just not document them for the regular embedding functions?

Really, a lot of those plotting functions could be more modular.

**Comment by flying-sheep** [comment]
Yeah. If nobody picks up the slack until I’m done with my thesis, I’ll do it after. But it’s really a thing we need.

**Comment by ivirshup** [comment]
Could we make sure that those arguments don't get documented in a release? Also, I think the doc builds have been broken since the wishbone PR. It might be worth asking read the docs to enable [doc builds for our PRs](https://blog.readthedocs.com/building-docs-for-pull-requests/).
